src/auth-guard/auth-guard.spec.ts

```diff
@@ -0,0 +1,40 @@
+import { TestBed, inject } from '@angular/core/testing';
+import { FirebaseApp, AngularFireModule } from '@angular/fire';
+import { COMMON_CONFIG } from './test-config';
+import { AngularFireAuthModule } from '@angular/fire/auth';
+import { AngularFireAuthGuardModule, AngularFireAuthGuard } from '@angular/fire/auth-guard';
+import { RouterModule, Router } from '@angular/router';
+import { APP_BASE_HREF } from '@angular/common';
+
+describe('AngularFireAuthGuard', () => {
+    let app: FirebaseApp;
+    let router: Router;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [
+                AngularFireModule.initializeApp(COMMON_CONFIG),
+                AngularFireAuthModule,
+                AngularFireAuthGuardModule,
+                RouterModule.forRoot([
+                    { path: 'a', redirectTo: '/', canActivate: [AngularFireAuthGuard] }
+                ])
+            ],
+            providers: [
+                { provide: APP_BASE_HREF, useValue: 'http://localhost:4200/' }
+            ]
+        });
+        inject([FirebaseApp, Router], (app_: FirebaseApp, router_: Router) => {
+            app = app_;
+            router = router_;
+        })();
+    });
+
+    afterEach(done => {
+        app.delete().then(done, done.fail);
+    });
+    
+    it('should be injectable', () => {
+        expect(router).toBeTruthy();
+    });
```

Comment:
[here's a StackOverflow answer](https://stackoverflow.com/a/53881037/4816918) with suggestions on ways to test auth guards. But maybe that's overkill and we just have a dummy user object and quick happy path tests for the different built-in pipes for now?

Reply:
Maybe also spy on the AngularFireAuthGuard pipe to make sure it calls the pipe provided to it with the data provided to it?

Reply:
I think I'll have to put together a proper E2E for this, something I'm working on on another branch. Think I'll punt on building more robust tests for now & take it as an action item.